aten/src/ATen/native/xnnpack/Pooling.cpp

```diff
@@ -0,0 +1,321 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/Pool.h>
+#include <ATen/native/utils/ParamUtils.h>
+#include <ATen/native/xnnpack/Common.h>
+#include <ATen/native/xnnpack/Factory.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace max_pool2d {
+
+struct Context final {
+  Operator max_pool_op;
+
+  struct Output final {
```

Comment:
nit: you'll probably need similar structure for AvgPool and other kinds of pooling, so it makes sens not to put it in max_pool2d namespace

Reply:
Thanks, I'll move it in a future patch after Kimish merges his.  The only place I can currently put it right now is inside Common.h which is starting to become a cluttered mess.

Reply:
OK, done now that Kimish's patch is merged.